### PR TITLE
[FIX] use a nonserialized cursor for updating user groups

### PR DESCRIPTION
**Pull request by hbrunn**
Doing this in a serialized transaction can cause deadlocks, and we don't need serialization here

### CODE REVIEW

File: auth_dynamic_groups/model/res_users.py [hunk 1]
```diff
@@ -35,10 +35,9 @@ def _login(self, db, login, password):
         return uid
 
     def update_dynamic_groups(self, uid, db):
-        pool = RegistryManager.get(db)
-        cr = pool._db.cursor()
-        user = pool.get('res.users').browse(cr, SUPERUSER_ID, uid)
-        groups_obj = pool.get('res.groups')
+        cr = self.pool._db.cursor(serialized=True)
```
Comment (moylop260) [inline auth_dynamic_groups/model/res_users.py]: Serialized=True to avoid serialized?

Comment (hbrunn) [inline auth_dynamic_groups/model/res_users.py]: @moylop260 thanks so much, I messed up the only thing that was to mess up here...

**Comment by pedrobaeza** [comment]
Are you sure you don't need to serialize? How user get correct groups then?

**Comment by hbrunn** [comment]
Using a nonserialized cursor doesn't always help, as the ORM also wants to update the user's `write_{uid,date}`, which blocks again, we do it in SQL

**Comment by hbrunn** [comment]
@pedrobaeza the worst thing that can happen is to have not all dynamic groups, which will be fixed with the next login

**Comment by moylop260** [comment]
Is posible add a test case that reproduce the problem?

**Comment by hbrunn** [comment]
difficult, because it's a timing issue with multiple users involved

**Comment by moylop260** [comment]
:+1:

**Comment by StefanRijnhart** [comment]
:+1:

**Pull request merged**